lib/SyTest/Federation/Server.pm

```diff
@@ -96,6 +103,34 @@ sub get_event
    return $event;
 }
 
+sub get_auth_chain
+{
+   my $self = shift;
+   my @event_ids = @_;
+
+   my %events_by_id = map { $_ => $self->get_event( $_ ) } @event_ids;
+
+   my @auth_chain = @event_ids;
+
+   while( @event_ids ) {
+      my $event = $events_by_id{shift @event_ids};
+
+      my @auth_ids = map { $_->[0] } @{ $event->{auth_events} };
```

Comment:
That would indeed be nice. :) However, it's a data structure baked into the Matrix federation protocol so there's not a lot we can do about that here.